profiler/internal/stackparse/stackparse_test.go

```diff
@@ -0,0 +1,372 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package stackparse
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+var update = flag.Bool("update", false, "update golden files")
+
+// TestParse_GoldenFiles verifies the output of the parser for a set of input
+// files in the test-fixtures. If you want to test a new behavior or bug fix,
+// this is the best place to do it.
+func TestParse_GoldenFiles(t *testing.T) {
```

Comment:
> tracebacks with cgo calls

Good idea. I added a test case for it in 3317f51bd3a562054079d3a7b694565e178645df - seems like there is no need to change the parser for it.

> traceback output from previous releases of go. The stack dump format isn't formally defined. Testing against a dump from 1.10 or 1.12 might be worth trying

I've [reviewed](https://gist.github.com/felixge/a136ef4d445ffe72b151c048188c15b2) the changes between go1.12 (the earliest version of Go supported by dd-trace-go, see README) and the latest release. Older versions of Go might print info about leftover defers, but I think it's okay to not support that.

But if you want I can try to create such a stack trace with an older version of Go so that we have an example of it in the test suite.

Reply:
I agree we don't need to support leftover defers, just wanted to make sure we'd vetted any changes from 1.12.  I figured your parser was robust enough to not break at least. This is perfect.